daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/dseltotql/DselToTqlVisitor.java

```diff
@@ -0,0 +1,263 @@
+        }
+    }
+
+    @Override
+    public TqlElement visitAssignment(ELNode elNode) {
+        /*
+         * `field_name=123` is considered as an assignment, despite this is the following which should be an assignment:
+         * `let field_name=123`. Finally we decided to consider this case as an equality.
+         */
+        return TqlBuilder.eq(elNode.getChild(0).getImage(), elNode.getChild(1).getImage());
+    }
+
+    @Override
+    public TqlElement visitIfThenElse(ELNode elNode) {
+        /*
+         * DSEL supports this since V1.1.0 but not yet TQL
+         */
+        throw new IllegalStateException("Unsupported conditional expression : " + elNode.getImage());
+    }
+
+    @Override
+    public TqlElement visitSwitchCases(ELNode elNode) {
+        /*
+         * DSEL supports this since V1.1.0 but not TQL
+         */
+        throw new IllegalStateException("Unsupported conditional expression : " + elNode.getImage());
+    }
+
+    private Expression[] buildExpressionsArray(ELNode elNode) {
+        List<Expression> list = new ArrayList<>();
+        for (ELNode child : elNode.getChildren()) {
+            Expression accept = (Expression) child.accept(this);
+            list.add(accept);
+        }
+        return list.toArray(new Expression[0]);
+    }
+
+    private TqlElement visitBetweenFunction(ELNode elNode) {
+        final TqlElement field = elNode.getChild(0).accept(this);
+        final ELNode leftParameter = elNode.getChild(1);
+        final ELNode rightParameter = elNode.getChild(2);
+
+        if (!isLiteral(leftParameter.getType().name()) || isLiteral(rightParameter.getType().name())) {
+            throw new IllegalArgumentException(
+                    "One or two of the parameters is not a literal, whereas the between function can only accept literals");
+        }
+
+        final LiteralValue literalValueOnLeft = (LiteralValue) visitLiteral(leftParameter);
+        final LiteralValue literalValueOnRight = (LiteralValue) visitLiteral(rightParameter);
+        return new FieldBetweenExpression(field, literalValueOnLeft, literalValueOnRight, false, false);
+    }
+
+    private TqlElement visitInFunction(ELNode elNode) {
+        final TqlElement field = elNode.getChild(0).accept(this);
+        List<ELNode> children = elNode.getChildren();
+
+        final LiteralValue[] literalValues = new LiteralValue[children.size() - 2];
+
+        for (int i = 1; i < children.size(); i++) {
+            ELNode child = children.get(i);
+
+            if (!isLiteral(child.getType().name())) {
+                throw new IllegalArgumentException(
+                        "At least one of the parameters is not a literal, whereas the in function can only accept literals");
+            }
+
+            literalValues[i - 1] = (LiteralValue) visitLiteral(child);
+        }
+
+        return new FieldInExpression(field, literalValues);
```

Comment:
The index shifting in this code is a bit mind-bending. 
Can we simplify this?

```suggestion
    private TqlElement visitInFunction(ELNode elNode) {
        final TqlElement field = elNode.getChild(0).accept(this);
        List<TqlElement> candidateValues = elNode.getChildren()
            .stream()
            .skip(1) // The first child is the field, the rest is the collection its value must belong to
            .map(child -> {
                if (!isLiteral(child.getType().name())) {
                    throw new IllegalArgumentException(
                            "At least one of the parameters is not a literal, whereas the in function can only accept literals");
                }

                return (LiteralValue) visitLiteral(child);
            })
            .collect(Collectors.toList());
            
        return new FieldInExpression(field, candidateValues);
```

Reply:
I just did a simplification which uses the idea of Cyprien :)